packages/server/src/database.test.ts

```diff
@@ -1,72 +1,99 @@
-import { deepClone } from '@medplum/core';
+import { deepClone, sleep } from '@medplum/core';
 import { EventEmitter } from 'node:events';
 import { Duplex } from 'node:stream';
 import pg, { Pool, PoolClient, PoolConfig, QueryArrayResult } from 'pg';
 import { Readable, Writable } from 'stream';
-import { MedplumDatabaseConfig, MedplumDatabaseSslConfig, loadConfig } from './config';
-import { closeDatabase, initDatabase } from './database';
-
-jest.mock('pg');
-
-const poolSpy = jest.spyOn(pg, 'Pool').mockImplementation((_config?: PoolConfig) => {
-  class MockPoolClient extends Duplex implements PoolClient {
-    release(): void {}
-    async connect(): Promise<void> {}
-    async query(): Promise<QueryArrayResult<any>> {
-      return {
-        command: '',
-        rowCount: null,
-        oid: -1,
-        fields: [],
-        rows: [],
-      };
-    }
-    copyFrom(_queryText: string): Writable {
-      return new Writable();
-    }
-    copyTo(_queryText: string): Readable {
-      return new Readable();
-    }
-    pauseDrain(): void {}
-    resumeDrain(): void {}
-    escapeIdentifier(_str: string): string {
-      return '';
-    }
-    escapeLiteral(_str: string): string {
-      return '';
-    }
-    getTypeParser(): any {
-      return undefined;
-    }
-    setTypeParser(): void {}
-  }
-
-  class MockPool extends EventEmitter implements Pool {
-    totalCount = -1;
-    idleCount = -1;
-    waitingCount = -1;
-    async connect(): Promise<pg.PoolClient> {
-      return new MockPoolClient();
-    }
-    on(): this {
-      return this;
-    }
-    async end(): Promise<void> {}
-    async query(): Promise<QueryArrayResult<any>> {
-      return {
-        command: '',
-        rowCount: null,
-        oid: -1,
-        fields: [],
-        rows: [],
-      };
-    }
-  }
-
-  return new MockPool();
-});
+import { MedplumDatabaseConfig, MedplumDatabaseSslConfig, loadConfig, loadTestConfig } from './config';
+import {
+  acquireAdvisoryLock,
+  closeDatabase,
+  DatabaseMode,
+  getDatabasePool,
+  initDatabase,
+  releaseAdvisoryLock,
+} from './database';
 
 describe('Database config', () => {
+  let poolSpy: jest.SpyInstance<pg.Pool, [config?: pg.PoolConfig | undefined]>;
+  beforeAll(() => {
+    jest.mock('pg');
```

Comment:
Hmm I see these tests broke, I wonder if we still need to mock at the top of the file, since `pg` is imported in some modules that are imported before the mock is done... 🤔 Just a hunch

In my data migration PR I basically just started a new file for anything that didn't need to mock `pg` to avoid dealing with this now that I remember